### ISSUE TITLE
Implement record mode

### ISSUE THREAD

**Issue by indigo0086** [t9md/atom-vim-mode-plus#238]
vim has the ability to record macros with q<char> and play them back with the @<char>.  It would add very useful functionality to the plugin.

**Comment by t9md** [comment]
I know and I can implement it, but I don't want to for now.
Until I feel I can manage complex situation brought by Macro feature, I will postpone to provide macro.

**Comment by expelledboy** [comment]
You should see the following:
https://github.com/JunSuzukiJapan/atom-keyboard-macros
https://github.com/JunSuzukiJapan/atom-keyboard-macros-vim
https://github.com/JunSuzukiJapan/atom-keyboard-macros-vim/issues/5

**Comment by t9md** [comment]
Why? You can do it by yourself.

**Comment by expelledboy** [comment]
I have just started using Atom, about two days ago, having used vim for years.

Was just pointing those interested in the issue to an existing implementation,
which doesn’t actually work very well with vim-mode… but it might be a start?

If you don’t get to it in time then in about 2 months I will do it myself (actually very interested in Electron).

**Comment by t9md** [comment]
OK, I think I won't implement macro feature within 2months or more.
It's interesting if you can create it.
For vim-mod-plus, I've released several packages as vim-mode-plus's plugin.

Wiki section have short instruction and list of vmp plugin packages for reference.

**Comment by sahil-a** [comment]
+1

**Comment by rtfeldman** [comment]
PSA: this worked for me - got me macros in `vim-mode-plus`!

https://github.com/JunSuzukiJapan/atom-keyboard-macros-vim/issues/5#issuecomment-212849566 via @expelledboy

**Comment by avindra** [comment]
All you need to do is install `atom-keyboard-macros-vim` now. The author merged in `vim-mode-plus` support

https://github.com/JunSuzukiJapan/atom-keyboard-macros-vim/commit/4804fb4a67a7f325814d83f470225260912c1364

**Comment by michaelsuo** [comment]
btw: `atom-keyboard-macros` doesn't fully support `vim-mode-plus`. Seems like a bunch of motions/commands are not supported. For example, `df<something>` will not be recorded in a macro.

**Comment by dylnmc** [comment]
I agree that atom-keyboard-macros does not provide extensive functionality, and it would be nice—if t9 gets a change—to provide macro replay/recording.

**Comment by johannesschweig** [comment]
Somebody made a package a month ago for that (vim-mode-plus-macros).

**Comment by mwielondek** [comment]
PSA: all the current plugins mentioned in this thread are terribly broken to the degree I wouldn't bother with them. Any chance of revisiting this, @t9md?
